src/arch/x86/vm.c

```diff
@@ -131,7 +132,9 @@ int vm_arch_load_image(vm_t *v, void *data, size_t datasz)
     void *kernel = ((uint8_t *) v->mem) + 0x100000;
 
     memset(boot, 0, sizeof(struct boot_params));
-    memmove(boot, data, sizeof(struct boot_params));
+    memmove((void *) boot + offsetof(struct boot_params, hdr),
```

Comment:
When using void pointers in calculations, the behavior is undefined.
Consider the following:
```c
    memmove((void *) ((uintptr_t) boot + offsetof(struct boot_params, hdr)),
            (void *) ((uintptr_t) data + offsetof(struct boot_params, hdr)),
            sizeof(struct setup_header));
```

Reply:
Add some comments to explain the rationale.